ddtrace/utils/formats.py

```diff
@@ -97,7 +101,29 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    if _TAGS_NOT_SEPARATED.search(tags_str):
```

Comment:
what about `"key:value"` ?

This regex will cause that use case to not work.

Reply:
The regex is for two colons without a valid separator in between. There is a test case for the simple case you gave.

Reply:
oh wait... I think I completely misread this... this is looking to see if there is more than 1 tag without a space or comma?

Reply:
yep, like `"key2:val1:"` test

Reply:
why do we need this check? does it help short circuit our parsing algorithm so we don't waste time parsing something that shouldn't, or does it catch an edge case that our parsing algorithm doesn't catch?

Reply:
it's an edge case that would otherwise be treated as a problem with separators, which it might not be the case, e.g., `key:val key2:val_with_a_colon_:`.